helpers_test.go

```diff
@@ -0,0 +1,437 @@
+		t.Fatalf("touch(%q): %s", filepath.Join(path...), err)
+	}
+	if shouldWait(path...) {
+		eventSeparator()
+	}
+}
+
+// mv
+func mv(t *testing.T, src string, dst ...string) {
+	t.Helper()
+	if len(dst) < 1 {
+		t.Fatalf("mv: dst must have at least one element: %s", dst)
+	}
+
+	var err error
+	switch runtime.GOOS {
+	case "windows", "plan9":
+		err = os.Rename(src, filepath.Join(dst...))
+	default:
+		err = exec.Command("mv", src, filepath.Join(dst...)).Run()
+	}
+	if err != nil {
+		t.Fatalf("mv(%q, %q): %s", src, filepath.Join(dst...), err)
+	}
+	if shouldWait(dst...) {
+		eventSeparator()
+	}
+}
+
+// rm
+func rm(t *testing.T, path ...string) {
+	t.Helper()
+	if len(path) < 1 {
+		t.Fatalf("rm: path must have at least one element: %s", path)
+	}
+	err := os.Remove(filepath.Join(path...))
+	if err != nil {
+		t.Fatalf("rm(%q): %s", filepath.Join(path...), err)
+	}
+	if shouldWait(path...) {
+		eventSeparator()
+	}
+}
+
+// rm -r
+func rmAll(t *testing.T, path ...string) {
+	t.Helper()
+	if len(path) < 1 {
+		t.Fatalf("rmAll: path must have at least one element: %s", path)
+	}
+	err := os.RemoveAll(filepath.Join(path...))
+	if err != nil {
+		t.Fatalf("rmAll(%q): %s", filepath.Join(path...), err)
+	}
+	if shouldWait(path...) {
+		eventSeparator()
+	}
+}
+
+// chmod
+func chmod(t *testing.T, mode fs.FileMode, path ...string) {
+	t.Helper()
+	if len(path) < 1 {
+		t.Fatalf("chmod: path must have at least one element: %s", path)
+	}
+	err := os.Chmod(filepath.Join(path...), mode)
+	if err != nil {
+		t.Fatalf("chmod(%q): %s", filepath.Join(path...), err)
+	}
+	if shouldWait(path...) {
+		eventSeparator()
+	}
+}
+
+// Collect all events in an array.
+//
+// w := newCollector(t)
+// w.collect(r)
+//
+// .. do stuff ..
+//
+// events := w.stop(t)
+type eventCollector struct {
+	w      *Watcher
+	events Events
+	mu     sync.Mutex
+	done   chan struct{}
+}
+
+func newCollector(t *testing.T) *eventCollector {
+	return &eventCollector{w: newWatcher(t), done: make(chan struct{})}
+}
+
+func (w *eventCollector) stop(t *testing.T) Events {
+	waitForEvents()
+
+	go func() {
+		err := w.w.Close()
+		if err != nil {
+			t.Error(err)
+		}
+	}()
+
+	select {
+	case <-time.After(1 * time.Second):
+		t.Fatal("event stream was not closed after 1 second")
+	case <-w.done:
+	}
+
+	w.mu.Lock()
+	defer w.mu.Unlock()
+	return w.events
+}
+
+func (w *eventCollector) collect(t *testing.T) {
+	go func() {
+		for {
+			select {
+			case e, ok := <-w.w.Errors:
+				if !ok {
+					w.done <- struct{}{}
+					return
+				}
+				t.Error(e)
+				return
+			case e, ok := <-w.w.Events:
+				if !ok {
+					w.done <- struct{}{}
+					return
+				}
+				w.mu.Lock()
+				w.events = append(w.events, e)
+				w.mu.Unlock()
+			}
+		}
+	}()
+}
+
+type Events []Event
+
+func (e Events) String() string {
+	b := new(strings.Builder)
+	for i, ee := range e {
+		if i > 0 {
+			b.WriteString("\n")
+		}
+		fmt.Fprintf(b, "%-20s %q", ee.Op.String(), strings.ReplaceAll(ee.Name, `\`, "/"))
```

Comment:
filepath.ToSlash ?

Reply:
Ah cheers, I didn't know about that function. Updated it.